go/status/utils.go

```diff
@@ -527,3 +532,60 @@ func GetFirstClient(v []keybase1.ClientStatus, typ keybase1.ClientType) *keybase
 	}
 	return nil
 }
+
+// zipLogs takes a slice of logs and returns `numBytes` worth of the
+// most recent log lines from the sorted set of all lines across all
+// logs.  The lines in each log should already be sorted, and the logs
+// should be sortable alphabetically between each other (e.g., each
+// line in every log should start with a timestamp in the same
+// format).
+func zipLogs(numBytes int, logs ...string) (res string) {
+	scanners := make([]*bufio.Scanner, len(logs))
+	for i, log := range logs {
+		scanners[i] = bufio.NewScanner(strings.NewReader(log))
+	}
+
+	// nextLines is the next chronological line in each log.  It's
+	// empty if we need to get another line from the log.
+	nextLines := make([]string, len(logs))
+	var buf bytes.Buffer
+	for {
+		// Fill in the next line.
+		minLine := 0
+		for i, currLine := range nextLines {
+			if currLine == "" {
+				for scanners[i].Scan() {
+					nextLines[i] = scanners[i].Text()
+					if nextLines[i] != "" {
+						break
+					}
+				}
+			}
+
+			// If we still don't have a line, just skip this log.
+			if nextLines[i] == "" {
+				continue
+			}
+
+			if minLine != i && nextLines[i] < nextLines[minLine] {
+				minLine = i
+			}
+		}
+
+		line := nextLines[minLine]
+		if line == "" {
+			// We're done!
+			break
+		}
+
+		buf.WriteString(line + "\n")
+		nextLines[minLine] = ""
+	}
+
+	if buf.Len() > numBytes {
```

Comment:
Should this be inside of the loop?

Reply:
No, because we need the _most recent_ `numBytes`.  If it were inside the loop, we'd break before we got to the end of the logs.

Reply:
🤦‍♂ right, misread the slicing